src/java/us/kbase/templates/module_method_spec_json.vm.properties

```diff
@@ -142,29 +142,25 @@
         "service-mapping": {
             "url": "",
             "name": "$module_name",
-            "method": "your_method",
+            "method": "run_$module_name",
             "input_mapping": [
                 {
                     "narrative_system_variable": "workspace",
-                    "target_argument_position": 0
+                    "target_property": "workspace_name"
```

Comment:
Ideally this would be a workspace ID, not a mutable name.

Reply:
I think this can be:
```
{
  "target_property": "workspace_id",
  "narrative_system_variable": "workspace_id"
}
```
You can pass this into KBaseReport as long as it's the python version (you added that to the perl version too i think?). See readme on python branch: https://github.com/kbaseapps/KBaseReport/tree/python

Reply:
I suppose I can switch the example code over to ws_id too

Reply:
@jayrbolton Did we ever get the python version deployed in prod?

Reply:
So, AssemblyUtils only takes workspace_name. I'd like these things to be consistent and I think having both would be more confusion that it's worth

Reply:
KBaseReport is only in CI still I think.
Have you ever heard of anyone encountering a problem in practical usage with workspace_name, or is it more of a theoretical benefit? If it has never caused a real problem, maybe workspace name is the better choice here. But also it probably wouldn't be hard to update AssemblyUtils

Reply:
It's pretty rare that workspace names change (especially for narratives), but it's possible. I'm more concerned that we're putting bad programming practices in our example boilerplate, and devs might continue those bad practices where it does matter more, like in object names.

Reply:
So our Utils are a bit of a mess on this. I looked though the APIs to see what would have to change if we wanted consistency.

Only WS ID: GenomeFileUtil, FeatureSetUtil, Generics Util, AssemblyUtil, fba_tools
Both params: KBaseReport, ReadsUtils
Destination ref (effectively both): ReadsAlignmentUtil, DifferentialExpressionUtil, ExpressionUtil
Only WS name: DataFileUtil, TreeUtils

So we might want to have a discussion on what the ideal save API would be. (I'm leaning toward using destination_ref where possible)